src/olympia/devhub/tasks.py

```diff
@@ -143,17 +145,21 @@ def wrapper(task, id_, hash_, *args, **kw):
 
 
 @validation_task
-def validate_file_path(path, hash_, listed, **kw):
+def validate_file_path(path, hash_, listed, is_webextension=False, **kw):
     """Run the validator against a file at the given path, and return the
     results.
 
     Should only be called directly by ValidationAnnotator."""
-
+    run_linter = (
+        waffle.switch_is_active('addons-linter')
+        and waffle.switch_is_active('webextensions'))
+    if is_webextension and run_linter:
+        return run_addons_linter(path, listed=listed)
     return run_validator(path, listed=listed)
 
 
 @validation_task
-def validate_file(file_id, hash_, **kw):
+def validate_file(file_id, hash_, is_webextension, **kw):
```

Comment:
`validate_file_path` signature is similar but provides a default value for `is_webextension`, might be better ? On a similar note, I personally hate functions with more than 2 non-keyword arguments, makes it hard to see at a quick glance what the arguments mean from the caller point of view.